src/Npgsql.NodaTime/Internal/LegacyTimestampTzHandler.cs

```diff
@@ -1,121 +0,0 @@
-﻿using System;
-using NodaTime;
-using NodaTime.TimeZones;
-using Npgsql.BackendMessages;
-using Npgsql.Internal;
-using Npgsql.Internal.TypeHandling;
-using Npgsql.PostgresTypes;
-using BclTimestampTzHandler = Npgsql.Internal.TypeHandlers.DateTimeHandlers.TimestampTzHandler;
-using static Npgsql.NodaTime.Internal.NodaTimeUtils;
-
-namespace Npgsql.NodaTime.Internal;
-
-sealed partial class LegacyTimestampTzHandler : NpgsqlSimpleTypeHandler<Instant>, INpgsqlSimpleTypeHandler<ZonedDateTime>,
-    INpgsqlSimpleTypeHandler<OffsetDateTime>, INpgsqlSimpleTypeHandler<DateTimeOffset>, 
-    INpgsqlSimpleTypeHandler<DateTime>, INpgsqlSimpleTypeHandler<long>
-{
-    readonly IDateTimeZoneProvider _dateTimeZoneProvider;
-    readonly TimestampTzHandler _wrappedHandler;
-
-    public LegacyTimestampTzHandler(PostgresType postgresType)
-        : base(postgresType)
-    {
-        _dateTimeZoneProvider = DateTimeZoneProviders.Tzdb;
-        _wrappedHandler = new TimestampTzHandler(postgresType);
-    }
-
-    #region Read
-
-    public override Instant Read(NpgsqlReadBuffer buf, int len, FieldDescription? fieldDescription = null)
-        => _wrappedHandler.Read(buf, len, fieldDescription);
-
-    ZonedDateTime INpgsqlSimpleTypeHandler<ZonedDateTime>.Read(NpgsqlReadBuffer buf, int len, FieldDescription? fieldDescription)
-    {
-        try
-        {
-            var instant = Read(buf, len, fieldDescription);
-
-            if (!DisableDateTimeInfinityConversions && (instant == Instant.MaxValue || instant == Instant.MinValue))
-                throw new InvalidCastException("Infinity values not supported for timestamp with time zone");
-
-            return instant.InZone(_dateTimeZoneProvider[buf.Connection.Timezone]);
```

Comment:
Uff this is some uglyness

Reply:
You mean the looking at the current TimeZone value for the connection? Yeah, it's not great but there are other cases of connection state which should conceivably impact how converters do their work...